enhancements/installer/assisted-installer-bare-metal-validations.md

```diff
@@ -0,0 +1,228 @@
+the bootstrap VM is launched, though, so it should be relatively quick
+in the overall installation process.
+
+There are no security implications for this change.
+
+## Design Details
+
+### Test Plan
+
+Besides the existing and new unit tests, the existing e2e tests for
+the `baremetal` platform will continue to test the full set of
+validation rules. The test suite for the assisted installer will
+verify that no extra validation rules are applied.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Alternatives
+
+### Add a new platform
+
+We could add a new platform to the installer to provide a clear home
+for all of the logic that is specific to the assisted installer. This
+would be a more invasive change, and might end up being confusing
+since the installer would either still need to configure the cluster
+with the `baremetal` platform or the `machine-api-operator` would need
+to understand a new platform and configure the correct machine API
+implementation.
+
+### Add flag(s) to install-config.yml
+
+We could add a flag or other input field to `install-config.yml` to
+control the validation behavior. For example,
+<https://github.com/openshift/installer/pull/3794> proposes a list of
+validation rules to skip. There are two reasons not to do this. First,
+it would expose the validation rules through the installer's public
+API, even though we would only want the assisted installer to use the
+feature. Second, it places the burden of managing the set of
+validations on the wrapper in the assisted installer, instead of on
+the author of validation rules in the installer source code. We know
+which validations should apply in each case, so there is no need to
+make the validations configurable.
+
+### Add an environment variable
```

Comment:
Pretty sure this is a non-starter. The installer team will reject this idea based on previous experience.

Reply:
I agree, that's why it's down here in the alternatives list.